dojo/tools/veracode_pipeline/parser.py

```
@@ -0,0 +1,95 @@
+import json
+from dojo.models import Finding
+
+
+class VeracodePipelineParser(object):
+    """This parser is written for Veracode Pipeline JSON output.
+
+    For details about the Veracode Pipeline Scan
+    see https://help.veracode.com/r/t_run_pipeline_scan
+    """
+
+    vc_severity_mapping = {
+        1: 'Info',
+        2: 'Low',
+        3: 'Medium',
+        4: 'High',
+        5: 'Critical'
+    }
+
+    def get_scan_types(self):
+        return ["Veracode Pipeline Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Veracode Pipeline Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Veracode Pipeline Scan Results"
+
+    def get_findings(self, filename, test):
+        if filename is None:
+            return
```

```suggestion
```
It will silently swallow an error (filename is never null for this parser)